### ISSUE TITLE
Perform automated feature engineering with evolutionary algorithms

### ISSUE THREAD

**Issue by bogoconic1** [bogoconic1/Qgentic-AI#91]
The current version of Qgentic-AI is incapable of performing extensive feature engineering.

What this limitation imposes: In normal tabular competitions, Qgentic-AI performs better with a neural network based solution (which automatically does feature engineering) than gradient boosted trees. It is unlikely to perform as well if extensive manual feature engineering is the silver bullet to winning the competition. [HIGH IMPACT]

It’s very expensive to AB test features one by one, plus I want it to find features that humans will not think of - figured that evolutionary algorithms may be the way to go

Cite: https://arxiv.org/pdf/2510.26144

**Comment by JasonZhangHub** [comment]
1. I do not think A/B test features with full training loop. We can consider using proxy metrics (simple ridge regression or single pass decision tree to score new feature potential before fully committing to it.
2. we can consider LLM-driven feature engineering. Feed the schema and column descriptions to an LLM and let the LLM suggest derived features that might be predictive.
